webapp/channels/src/components/admin_console/search_keyword_marking.tsx

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import debounce from 'lodash/debounce';
+import Mark from 'mark.js';
+import React, {useEffect, useMemo, useRef} from 'react';
+import type {ReactNode} from 'react';
+
+type Props = {
+    keyword?: string;
+    pathname?: string;
+    children: ReactNode;
+}
+
+const DEBOUNCE_WAIT_TIME = 300;
+
+function SearchKeywordMarking(props: Props) {
+    const containerRef = useRef<HTMLDivElement>(null);
+    const markJsRef = useRef<Mark>();
+
+    function doMark(keyword: string) {
+        markJsRef.current = new Mark(containerRef.current as HTMLDivElement);
```

Comment:
If you define this inside `debouncedRedrawHighlight` after checking if `!containerRef.current`, you won't need the type assertion.

Alternatively, you could also pass `containerRef.current` into this as an argument. If you do, TS will know that it's not undefined because of that null check, so you also wouldn't need the type assertion if you do it that way

Reply:
handled